mesonbuild/modules/python3.py

```diff
@@ -36,7 +46,8 @@ def __init__(self, *args, **kwargs):
         })
 
     @permittedKwargs(known_shmod_kwargs)
-    def extension_module(self, state, args, kwargs):
+    @typed_kwargs('python3.extension_module', *_MOD_KWARGS, allow_unknown=True)
+    def extension_module(self, state, args, kwargs: SharedModuleKW):
```

Comment:
1) why allow_unknown?
2) if you remove allow_unknown this could get rid of the below error handling, which we don't do in the maintained python module -- IMHO there isn't real value in customizing the error message here, it doesn't improve the end user's understanding of why not to use it.

Reply:
ignore

Reply:
The goal is to get there, I'm basically just trying to maximize the profit (and coverage) of new arguments to the build targets by covering the module wrappers at the same time. So once `name_prefix` and `name_suffix` are handled by the decorator, yeah, those will just disapear

Reply:
I just removed them from the `permittedKwargs` call, which will allow their removal from the body